### PR TITLE
Use query providers that are DICOM

### PR DESCRIPTION
**Pull request by funcionante**
This targets the last update in #122. Two things that might be important to note:
- I used Java 8; since we are not supporting Java 7 anymore, I thought it would be ok;
- In `GET searchDIM/` I am ignoring providers that are not valid, instead of throwing an error; however, the web app is handling that, when using a multi-select that validates the providers; is it ok?

Also, possibly I forgot to apply the new settings somewhere. Please tell me if something is missing.

### CODE REVIEW

File: dicoogle/src/main/java/pt/ua/dicoogle/server/web/servlets/search/DumpServlet.java [hunk 1]
```diff
@@ -58,11 +55,13 @@ protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws Se
...
         }
-        
+
         String[] providerArr = req.getParameterValues("provider");
-        List<String> providers = (providerArr == null)
-                ? PluginController.getInstance().getQueryProvidersName(true)
-                : Arrays.asList(providerArr);
+        List<String> providers = PluginController.getInstance().filterDicomQueryProviders(providerArr);
+
+        if (providers.size() == 0) {
```
Comment (Enet4) [inline dicoogle/src/main/java/pt/ua/dicoogle/server/web/servlets/search/DumpServlet.java]: I was initially thinking about only using the internal dimProvider list if the provider parameter was empty. But I suppose this works too. We'll just make sure that both Bioinformatics and BMD agree on this API.

File: dicoogle/src/main/java/pt/ua/dicoogle/server/web/servlets/search/DumpServlet.java [hunk 2]
```diff
@@ -58,11 +55,13 @@ protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws Se
...
+            resp.sendError(400, "No DIM providers supplied.");
```
Comment (Enet4) [inline dicoogle/src/main/java/pt/ua/dicoogle/server/web/servlets/search/DumpServlet.java]: This servlet could use the same fix as the one in `/search`, but I guess we can postpone this.

**Comment by funcionante** [comment]
Thank you for the feedback. I applied the changes you proposed to `filterDicomQueryProviders()`.

**Comment by bastiao** [comment]
@Enet4 there is only one dim provider supported in settings? We will need more than one. Any reason why only single plugin is supported at settings? 

There is also missing the DICOM Q/R part, but I can handle.

**Comment by Enet4** [comment]
@bastiao Yes, it should work for multiple DIM providers. What you seem to have found is a bug in the server settings deserialization process, which should have admitted multiple `<dim-provider>` elements into the same list. I am currently looking into that.

**Comment by bastiao** [comment]
@Enet4 yap, it should be

```
<dim-providers>
<dim-provider>a</dim-provider>
<dim-provider>b</dim-provider>
...
<dim-provider>n</dim-provider>
</dim-providers>
```

**Pull request merged**